diff --git a/arch/arm64/boot/dts/rockchip/overlay/Makefile b/arch/arm64/boot/dts/rockchip/overlay/Makefile
new file mode 100644
index 000000000..5bc873521
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/Makefile
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: GPL-2.0
+dtbo-$(CONFIG_ARCH_ROCKCHIP) += \
+       rockchip-rk3588-sata.dtbo
+
+dtb-y += $(dtbo-y)
+
+clean-files    := *.dtbo
diff --git a/arch/arm64/boot/dts/rockchip/overlay/rockchip-rk3588-sata.dtso b/arch/arm64/boot/dts/rockchip/overlay/rockchip-rk3588-sata.dtso
new file mode 100644
index 000000000..2759ab9cf
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/rockchip-rk3588-sata.dtso
@@ -0,0 +1,36 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+       fragment@0 {
+               target = <&pcie2x1l1>;
+
+               __overlay__ {
+                       status = "disabled";
+               };
+       };
+
+       fragment@1 {
+               target = <&pcie2x1l2>;
+
+               __overlay__ {
+                       status = "disabled";
+               };
+       };
+
+       fragment@2 {
+               target = <&sata0>;
+
+               __overlay__ {
+                       status = "okay";
+               };
+       };
+
+       fragment@3 {
+               target = <&sata2>;
+
+               __overlay__ {
+                       status = "okay";
+               };
+       };
+};
